oper8/deploy_manager/replace_utils.py

```diff
@@ -0,0 +1,128 @@
+"""This file contains common utilities for detecting if a replace operation is required
+for a resource
+"""
+# Standard
+from typing import Any, Callable, List
+
+# Third Party
+from openshift.dynamic.apply import recursive_diff, recursive_list_diff
+
+# First Party
+import alog
+
+log = alog.use_channel("DMRPLC_UTILS")
+
+
+def modified_lists(
+    current_manifest: dict, desired_manifest: dict, in_list: bool = False
+) -> bool:
+    """Helper function to check if there are any differences in the lists of the desired manifest.
+    This is required because Kubernetes combines lists which is often not the desired use
+    """
+    # If type mismatch then assume replace
+    if (
+        desired_manifest
+        and current_manifest
+        and type(desired_manifest) is not type(current_manifest)
```

Comment:
The one thought on this `is` comparison is that it would not allow for subclasses (e.g. `dict` and `aconfig.Config`). I think that should be OK, though, right?

Reply:
It should be okay because this is only called after the component rendering which should convert all of those objects to dicts/native types

Reply:
Makes sense. Ship it!